test/e2e/adminapi_cluster_getlogs.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+var _ = Describe("[Admin API] Kubernetes get pod logs action", func() {
+	BeforeEach(skipIfNotInDevelopmentEnv)
+
+	const containerName = "e2e-test-container-name"
+	const podName = "e2e-test-pod-name"
+
+	When("in a standard openshift namespace", func() {
+		const namespace = "default"
+
+		It("should be able to get logs from a container of a pod", func() {
+			ctx := context.Background()
+			testGetPodLogsOK(ctx, containerName, podName, namespace)
+		})
+	})
+
+	When("in a customer namespace", func() {
+		const namespace = "e2e-test-namespace"
+
+		It("should be not be able to get logs from customer workload namespaces", func() {
+			ctx := context.Background()
+			testGetPodLogsFromCustomerNamespaceForbidden(ctx, containerName, podName, namespace)
+		})
+	})
+})
+
+// We will create a pod with known logs of its container and will compare the logs gotten through the kubernetes client and through the Admin API.
+func testGetPodLogsOK(ctx context.Context, containerName, podName, namespace string) {
+	By("creating a pod in default namespace with some known logs then comparing those logs with the logs of the container in the pod created")
+	pod := mockPod(containerName, podName, namespace)
+	pod, err := clients.Kubernetes.CoreV1().Pods(namespace).Create(ctx, pod, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		err = clients.Kubernetes.CoreV1().Pods(namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	}()
+	err = wait.Poll(time.Second, time.Second*60, func() (done bool, err error) {
+		pod, err = clients.Kubernetes.CoreV1().Pods(namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		Expect(err).NotTo(HaveOccurred())
+		if pod.Status.Phase == corev1.PodSucceeded {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	if err == wait.ErrWaitTimeout {
```

Comment:
not understanding why this isn't just 
```suggestion
Expect(err).NotTo(HaveOccurred())
```
I think we can delete all this err analyzing logic through line 74

Reply:
Thought of throwing a Clouderror in case of Timeout but Yaa .. I agree with you .. it can be just Expect() .. I will change this ..

Reply:
Clouderrors are mostly for production error handling, this is just our internal e2e testing, the more error details the better because if it fails we will be responsible for investigating that failure.

Reply:
Okay .. gotcha! .. I anyways .. made changes in my last commit .. so you should see those changes ..